tests/components/homeworks/test_init.py

```diff
@@ -165,3 +166,25 @@ async def test_send_command(
             blocking=True,
         )
     assert len(mock_controller._send.mock_calls) == 0
+
+
+async def test_cleanup_on_ha_shutdown(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_homeworks: MagicMock,
+) -> None:
+    """Test cleanup when HA shuts down."""
+    mock_controller = MagicMock()
+    mock_homeworks.return_value = mock_controller
+
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    mock_homeworks.assert_called_once_with("192.168.0.1", 1234, ANY)
+    mock_controller.stop.assert_not_called()
+
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_STOP)
```

Comment:
Can't you also do `hass.config_entries.async_unload_entry(..)`?

Reply:
No, the integration has some specific code for shutdown to make sure the worker thread is stopped.

Reply:
Ah, this test is not related to the code change in async unload entry?

Reply:
Exactly, the point of that code change is to make async_unload_entry fully covered.
The new test makes sure the `cleanup` function is called.